DataSetExplorer/DataSetBuilder/FullDataSetBuilderService.cs

```diff
@@ -0,0 +1,70 @@
+using DataSetExplorer.DataSetSerializer;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+
+namespace DataSetExplorer
+{
+    class FullDataSetBuilderService : IFullDataSetBuilder
+    {
+        public IEnumerable<IGrouping<string, DataSetInstance>> GetAnnotatedInstancesGroupedBySmells(int? annotatorId)
+        {
+            var allAnnotatedInstances = new List<DataSetInstance>();
+
+            var projects = GetAnnotatedProjects();
+            foreach (var key in projects.Keys)
+            {
+                CodeModelFactory factory = new CodeModelFactory();
+                CaDETProject project = factory.CreateProjectWithCodeFileLinks(key.ToString());
+
+                var importer = new ExcelImporter(projects[key].ToString());
+                var annotatedInstances = importer.Import("Clean CaDET").GetAllInstances();
+
+                LoadAnnotators(ref annotatedInstances);
+                if (annotatorId != null) annotatedInstances = annotatedInstances.Where(i => i.IsAnnotatedBy((int)annotatorId)).ToList();
+                allAnnotatedInstances.AddRange(FillInstancesWithMetrics(annotatedInstances, project));
+            }
+            return allAnnotatedInstances.GroupBy(i => i.Annotations.ToList()[0].InstanceSmell.Value);
+        }
+
+        private List<DataSetInstance> FillInstancesWithMetrics(List<DataSetInstance> annotatedInstances, CaDETProject project)
+        {
+            return annotatedInstances.Select(i => {
+                i.MetricFeatures = project.GetMetricsForCodeSnippet(i.CodeSnippetId);
+                return i;
+            }).ToList();
+        }
+
+        private ListDictionary GetAnnotatedProjects()
+        {
+            ListDictionary projects = new ListDictionary(); // local repository path, annotations folder path
+            projects.Add("D:/ccadet/annotations/repos/BurningKnight", "D:/ccadet/annotations/annotated/BurningKnight");
+            projects.Add("D:/ccadet/annotations/repos/Core2D", "D:/ccadet/annotations/annotated/Core2d");
+            projects.Add("D:/ccadet/annotations/repos/jellyfin", "D:/ccadet/annotations/annotated/Jellyfin");
+            projects.Add("D:/ccadet/annotations/repos/OpenRA", "D:/ccadet/annotations/annotated/OpenRA");
+            projects.Add("D:/ccadet/annotations/repos/ShareX", "D:/ccadet/annotations/annotated/ShareX");
+            projects.Add("D:/ccadet/annotations/repos/ShopifySharp", "D:/ccadet/annotations/annotated/ShopifySharp");
+            projects.Add("D:/ccadet/annotations/repos/MonoGame", "D:/ccadet/annotations/annotated/MonoGame");
+            projects.Add("D:/ccadet/annotations/repos/Osu", "D:/ccadet/annotations/annotated/Osu");
+            return projects;
+        }
+
+        private void LoadAnnotators(ref List<DataSetInstance> annotatedInstances)
+        {
+            List<Annotator> annotators = new List<Annotator>()
+            {
+                new Annotator(1, 6, 1),
```

Comment:
Hardcoded? I realize this is not the focus of the PR, but you should leave a TODO comment to ensure you don't forget to resolve these issues later.